src/array.cr

```diff
@@ -1770,4 +1770,15 @@ class Array(T)
       end
     end
   end
+
+  # This method returns the i-th element. If i is equal to ary.size, it returns nil.
+  def bsearch(&block : T -> Int|Float|Bool?)
+    idx=bsearch_index(&block)
+    idx ? self[idx] : nil
```

Comment:
Thank you for continuing to check this pull request. But should the nil be avoided so much? Perhaps I should get an opinion from @MakeNowJust (also see #1871)